server/controllers/list.js

```diff
@@ -1030,14 +1031,19 @@ const updateListItem = (req, resp) => {
           : ActivityType.ITEM_RESTORE;
       }
 
+      if (isDeleted !== undefined) {
```

Comment:
What about `isArchived: true, isDeleted: false`? Then the `editedItemActivity` would be `ActivityType.ITEM_DELETE`. Is that expected?

Reply:
When `isArchived` is true we are archiving, when false we are restoring. So here it is expected.
```
  if (isArchived !== undefined) {
        itemToUpdate.isArchived = isArchived;
        editedItemActivity = isArchived
          ? ActivityType.ITEM_ARCHIVE
          : ActivityType.ITEM_RESTORE;
      }
```


For `isDeleted` case, you're right it can be done like below because there will be no false state for 'restoring from deleted'. 
```
  if (isDeleted) {
        itemToUpdate.isDeleted = isDeleted;
        editedItemActivity = ActivityType.ITEM_DELETE;
   }
```

Reply:
As this can not be undone?, I think you should expose `DELETE` endpoint to handle this case separately.

Reply:
Deleting item can not be undone 

I've prepared separated endpoint for this. Check next commit, please :)